src/profilePage/ProfilePage.tsx

```diff
@@ -1,22 +1,129 @@
+import { useQuery } from "@tanstack/react-query";
+import { useEffect, useState } from "react";
 import Calendar from "react-github-contribution-calendar";
 import {
   monthLabelAttributes,
   panelAttributes,
   panelColors,
   until,
-  values,
   weekLabelAttributes,
 } from "../Models/calendarHeatmap";
+import {
+  CommitData,
+  CommitItem,
+  LanguageData,
+  ProfileItem,
+  RepoItem,
+} from "../Models/interfaces";
 import Drawer from "../Navigation/Drawer";
+import { Route } from "../routes";
+import { getProfile, getRepos } from "./Api/profileApi";
 import QuickStats from "./QuickStats";
 import RepoCard from "./RepoCard";
 import StatsCarousel from "./StatsCarousel";
 import StatsRadial from "./StatsRadial";
 
+interface RouteParams {
+  profileId: string;
+}
+
 function ProfilePage() {
+  const [profile, setProfile] = useState<ProfileItem | null>(null);
+  const [repos, setRepos] = useState<RepoItem[]>([]);
+  const [repoNumber, setRepoNumber] = useState<number>(0);
+  const [commitsNumber, setCommitsNumber] = useState<number>(0);
+  const [dataReady, setDataReady] = useState(false);
+  const [result, setResult] = useState<LanguageData>({});
+  const [commitsReady, setCommitsReady] = useState(false);
+  const [commits, setCommits] = useState<CommitData>({});
+
+  const { profileId } = Route.useParams<RouteParams>();
+  const profileName = profileId;
+
+  useEffect(() => {
+    getProfile(profileName, setProfile);
+    getRepos(profileName, setRepos, setRepoNumber);
+  }, []);
+
+  //tanstack/react-query hook to fetch the users
+  useQuery({
+    queryKey: [profileName?.trim()],
+    queryFn: () => getProfile(profileName, setProfile),
+    enabled: false,
+  });
+
+  useQuery({
+    queryKey: ["Repos"],
+    queryFn: () => getRepos(profileName, setRepos, setRepoNumber),
+    enabled: false,
+  });
+
+  function giveMeLanguages(languages: LanguageData) {
+    setResult((prevResult) => {
+      // Create a copy of the previous state
+      const updatedResult = { ...prevResult };
+
+      // Update counts or add new languages
+      for (const [language, count] of Object.entries(languages)) {
```

Comment:
Isn't the second part of the tuple the index not the count?
Therefore adding the count won't just add one each time 🤔

Reply:
Or if not, why use Object.entries?